File: app/src/main/java/mobiledev/erickgomes/picpayapp/models/CreditCard.java
package mobiledev.erickgomes.picpayapp.models;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

/**
 * Created by erickgomes on 23/03/2018.
 */

@Entity(tableName = "creditcards")
public class CreditCard {
    private String card_owner;
    private String card_number;
    private String expiry_date;
    @PrimaryKey(autoGenerate = true)
    private int cvv;
    private int brand;
    private boolean isChecked;

    public CreditCard() {
        this.isChecked = false;
    }

    public boolean isChecked() {
        return isChecked;
    }

    public void setChecked(boolean checked) {
        isChecked = checked;
    }

    public String getCard_number() {
        return card_number;
    }


    public void setCard_number(String card_number) {
        this.card_number = card_number;
    }

    public String getExpiry_date() {
        return expiry_date;
    }

    public void setExpiry_date(String expiry_date) {
        this.expiry_date = expiry_date;
    }

    public int getCvv() {
        return cvv;
    }

    public void setCvv(int cvv) {
        this.cvv = cvv;
    }

    public int getBrand() {
        return brand;
    }

    public void setBrand(int brand) {
        this.brand = brand;
    }

    public String getCard_owner() {
        return card_owner;
    }

    public void setCard_owner(String card_owner) {
        this.card_owner = card_owner;
    }
}

File: app/src/main/java/com/v1pi/picpay_teste/Domains/CreditCard.kt
package com.v1pi.picpay_teste.Domains

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

@Entity
data class CreditCard(@PrimaryKey(autoGenerate = true) val uid: Int,
                      @ColumnInfo(name = "number") val number: String,
                      @ColumnInfo(name = "cvv") val cvv: Int,
                      @ColumnInfo(name = "expiry_date") val expiry_date: String) {
    override fun equals(other: Any?): Boolean {
        if(other is CreditCard) {
            return other.uid == this.uid && other.number == this.number && other.cvv == this.cvv && other.expiry_date == this.expiry_date
        }
        return false
    }

    override fun hashCode(): Int {
        var result = uid
        result = 31 * result + number.hashCode()
        result = 31 * result + cvv
        result = 31 * result + expiry_date.hashCode()
        return result
    }
}